### PR TITLE
Fix typos in Filters

### PR DESCRIPTION
**Pull request by rglsk**
Refer to #3109, done:
* Updated method and class docstring to follow the same convention (start with new line)
* Fix couple of typos
* Fix typing override issue with `check_update` method (@Bibo-Joshi : Please let me know if this change should be cherry-picked to other branch to merge across master -> I'm happy to do if if the proposed suggestion fits you 😄  )

### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs and all suitable `__all__` s

### CODE REVIEW

File: telegram/ext/filters.py [hunk 1]
```diff
@@ -182,8 +183,16 @@ def __init__(self, name: str = None, data_filter: bool = False):
         self._name = self.__class__.__name__ if name is None else name
         self._data_filter = data_filter
 
-    def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:  # skipcq: PYL-R0201
-        """Checks if the specified update is a message."""
+    def check_update(self, update: Update) -> bool:  # skipcq: PYL-R0201
```
Comment (harshil21) [inline telegram/ext/filters.py:186]: the type hint change should be reverted. The subclasses of `BaseFilter`s `check_update` can return `DataDict` based on `data_filter` property. That's why pylint also complained in `UpdateFilter` and `MessageFilter`

File: telegram/ext/filters.py [hunk 2]
```diff
@@ -153,7 +154,7 @@ class BaseFilter:
     If you want to create your own filters create a class inheriting from either
     :class:`MessageFilter` or :class:`UpdateFilter` and implement a ``filter()``
     method that returns a boolean: :obj:`True` if the message should be
-    handled, :obj:`False` otherwise.
+    handled, otherwise :obj:`False`.
```
Comment (harshil21) [inline telegram/ext/filters.py:157]: I think the previous one made more sense

**Review by harshil21 — CHANGES_REQUESTED** [review]

File: telegram/ext/filters.py (hunk 1 above)
Comment (rglsk) [reply telegram/ext/filters.py:186]: @harshil21 Are you sure? the base method can only return the True of False 🤔 The methods which overrides this one can return the `DataDict` 🤔 
Screenshot 2022-10-01 at 16 21 31

File: telegram/ext/filters.py (hunk 2 above)
Comment (rglsk) [reply telegram/ext/filters.py]: ✅ Done

File: telegram/ext/filters.py (hunk 1 above)
Comment (harshil21) [reply telegram/ext/filters.py:186]: imo, base method should have same sig as children (if possible). Let's see what others think

Comment (rglsk) [reply telegram/ext/filters.py]: @harshil21 I reverted the change for now, we can do it in another PR to get others opinions on this specific case later 👍

File: telegram/ext/filters.py [hunk 3]
```diff
@@ -183,7 +184,15 @@ def __init__(self, name: str = None, data_filter: bool = False):
...
     def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:  # skipcq: PYL-R0201
-        """Checks if the specified update is a message."""
+        """
+        Checks if the specified update is a message.
+
+        Args:
+            update (:class:`telegram.Update`): Incoming update.
+
+        Returns:
+            :obj:`bool`
```
Comment (harshil21) [inline telegram/ext/filters.py:194]:
```suggestion
            :obj:`bool` | :obj:`Dict[str, list]`
```

**Review by harshil21 — COMMENTED** [review]
let's also wait for other reviews. And, please avoid force pushing since it makes it harder to review

File: telegram/ext/filters.py [hunk 4]
```diff
@@ -183,7 +184,15 @@ def __init__(self, name: str = None, data_filter: bool = False):
...
     def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:  # skipcq: PYL-R0201
-        """Checks if the specified update is a message."""
+        """
+        Checks if the specified update is a message.
+
+        Args:
+            update (:class:`telegram.Update`): Incoming update.
+
+        Returns:
+            :obj:`bool` | :obj:`Dict[str, list]`
```
Comment (harshil21) [inline telegram/ext/filters.py:194]:
```suggestion
            :obj:`bool` | Dict[str, list]
```
sorry, forgot that the previous one fails the docs

File: telegram/ext/filters.py (hunk 1 above)
Comment (Bibo-Joshi) [reply telegram/ext/filters.py]: I agree with harshil here.

File: telegram/ext/filters.py [hunk 5]
```diff
@@ -183,7 +184,15 @@ def __init__(self, name: str = None, data_filter: bool = False):
...
     def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:  # skipcq: PYL-R0201
-        """Checks if the specified update is a message."""
+        """
+        Checks if the specified update is a message.
+
+        Args:
+            update (:class:`telegram.Update`): Incoming update.
+
+        Returns:
+            :obj:`bool` | Dict[str, list]
```
Comment (Bibo-Joshi) [inline telegram/ext/filters.py:194]:
```suggestion
            :obj:`bool` | Dict[:obj:`str`, :obj:`list`]
```

File: telegram/ext/filters.py [hunk 6]
```diff
@@ -116,7 +116,8 @@
 
 
 class BaseFilter:
-    """Base class for all Filters.
+    """
+    Base class for all Filters.
```
Comment (Bibo-Joshi) [inline telegram/ext/filters.py:120]: TBH I don't really see any value in these changes - in this PR or even at all, that is.
True, some style guides suggest to start the docstring on a new line, but none of the styling tools that we currently use will report on these things.  __If__ we want to have docstrings start on a new line, I would rather like to make it a standalone issue to apply this to the whole code base.
I'm inclined to ask to revert these changes. Let's first hear what others think, though.

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]

File: telegram/ext/filters.py (hunk 6 above)
Comment (harshil21) [reply telegram/ext/filters.py:120]: I also agree to revert, they are rendered the same and [PEP 257](https://peps.python.org/pep-0257/) does not comment on this design.

**Comment by Bibo-Joshi** [comment]
PS: you mentioned #3109 in all your PRs so far - I'd like to point out that none of your PRs addressed #3109 in any way 😉

**Comment by rglsk** [comment]
Alright folks, sorry for that, I'll try better next time 🙏

**Pull request closed without merging**